iothub/service/tests/DeviceAuthenticationTests.cs

```diff
@@ -51,7 +51,12 @@ public async Task DeviceAuthenticationGoodAuthConfigTest1()
             var mockHttpClient = new Mock<HttpClient>();
             mockHttpClient.Setup(restOp => restOp.SendAsync(It.IsAny<HttpRequestMessage>(), It.IsAny<CancellationToken>())).ReturnsAsync(mockHttpResponse);
 
-            var devicesClient = new DevicesClient(HostName, mockCredentialProvider.Object, mockHttpClient.Object, mockHttpRequestFactory);
+            var devicesClient = new DevicesClient(
+                HostName,
+                mockCredentialProvider.Object,
+                mockHttpClient.Object,
+                mockHttpRequestFactory,
+                new IotHubServiceRetryHandler(new IotHubServiceNoRetry()));
```

Comment:
Why not add this as a static class field and use it in all these cases? Unnecessary allocations.